src/rest_framework_dso/renderers.py

```diff
@@ -37,7 +40,41 @@ def setup_pagination(self, paginator: pagination.DelegatedPageNumberPagination):
         self.paginator = paginator
 
 
-class HALJSONRenderer(RendererMixin, JSONRenderer):
+class BrowsableAPIRenderer(RendererMixin, orig.BrowsableAPIRenderer):
+    def get_context(self, data, accepted_media_type, renderer_context):
+        context = super().get_context(data, accepted_media_type, renderer_context)
+
+        # Fix response content-type when it's filled in by the exception_handler
+        response = renderer_context["response"]
+        if response.content_type:
+            context["response_headers"]["Content-Type"] = response.content_type
+
+        return context
+
+    def render(self, data, accepted_media_type=None, renderer_context=None):
+        # Protect the browsable API from being used as DOS (Denial of Service) attack vector.
+        # While we allow infinitely large pages in our streaming responses, the browsable API
+        # still has to render that in-memory as part of the HTML template response.
+        request = renderer_context["request"]
+        view = renderer_context["view"]
+        if view.paginator.get_page_size(request) >= BROWSABLE_MAX_PAGE_SIZE:
```

Comment:
Dit is verwarrend: als 1000 het maximum is, dan verwacht ik 1000 te mogen opvragen (zo heb ik ook het commentaar hieronder geschreven). Het werkelijke maximum is echter 999. Check veranderen in `>`?

Reply:
goede, fixed!